dsl/src/main/scala/com.crobox.clickhouse/dsl/Query.scala

```diff
@@ -21,12 +24,53 @@ case object ASC extends OrderingDirection
 
 case object DESC extends OrderingDirection
 
-sealed case class InternalQuery(selectQuery: SelectQuery,
-                                from: FromQuery,
+sealed case class InternalQuery(select: Option[SelectQuery] = None,
+                                from: Option[FromQuery] = None,
                                 where: Option[Comparison] = None,
-                                groupBy: mutable.LinkedHashSet[AnyTableColumn] = mutable.LinkedHashSet.empty,
+                                groupBy: Seq[AnyTableColumn] = Seq.empty,
                                 having: Option[Comparison] = None,
                                 join: Option[JoinQuery] = None,
-                                orderBy: mutable.LinkedHashSet[(AnyTableColumn, OrderingDirection)] =
-                                    mutable.LinkedHashSet.empty,
-                                limit: Option[Limit] = None)
+                                orderBy: Seq[(AnyTableColumn, OrderingDirection)] = Seq.empty,
+                                limit: Option[Limit] = None) {
+
+  def isValid = {
+    val validGroupBy = groupBy.isEmpty && having.isEmpty || groupBy.nonEmpty
+
+    select.isDefined && validGroupBy
+  }
+
+  def isPartial = !isValid
+
+  /**
+   * Join 2 internalQueries, overwriting the other on conflicts
+   * @param other
+   */
+  def :+>(other: InternalQuery): InternalQuery =
+    InternalQuery(
+      select.orElse(other.select),
+      from.orElse(other.from),
+      where.orElse(other.where),
+      if (groupBy.nonEmpty) groupBy else other.groupBy,
+      having.orElse(other.having),
+      join.orElse(other.join),
+      if (orderBy.nonEmpty) orderBy else other.orderBy,
+      limit.orElse(other.limit)
+    )
+
+  def <+:(other: InternalQuery): InternalQuery = :+>(other)
```

Comment:
Add some documentation about what those expression means? Now it looks like sad santa...

Reply:
added